x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/inference/InferenceRunner.java

```diff
@@ -73,10 +73,12 @@ public void run(String modelId) {
         LOGGER.info("[{}] Started inference on test data against model [{}]", config.getId(), modelId);
         try {
             PlainActionFuture<LocalModel> localModelPlainActionFuture = new PlainActionFuture<>();
-            modelLoadingService.getModelForSearch(modelId, localModelPlainActionFuture);
+            modelLoadingService.getModelForPipeline(modelId, localModelPlainActionFuture);
             LocalModel localModel = localModelPlainActionFuture.actionGet();
             TestDocsIterator testDocsIterator = new TestDocsIterator(new OriginSettingClient(client, ClientHelper.ML_ORIGIN), config);
-            inferTestDocs(localModel, testDocsIterator);
+            try (localModel) {
```

Comment:
Would it make sense to change this line to:
```
try (LocalModel localModel = modelLoadingService.getModelForPipeline(modelId, localModelPlainActionFuture).actionGet()) {
```
?

Reply:
I did a slight refactor to take this comment into account. 

Ultimately, it does not really matter.